pkg/agent/agent.go

```diff
@@ -908,9 +912,27 @@ func (i *Initializer) initializeIPSec() error {
 			return fmt.Errorf("IPsec was requested, but the OVS IPsec monitor does not seem to be running")
 		}
 	}
-
-	if err := i.readIPSecPSK(); err != nil {
-		return err
+	if i.networkConfig.IPsecConfig.AuthenticationMode == config.IPsecAuthenticationModePSK {
+		if err := i.readIPSecPSK(); err != nil {
+			return err
+		}
+		// Clean up stale configs in OVS database.
```

Comment:
Do you think it is ideal to break if user changes cert auth back to psk auth without restarting Nodes? I think for now it works for other encryption mode changes.

Reply:
It's not ideal but are other mode changes supported currently? for example, now we have 4 encryption modes: non encryption, encryption with wireguard, encryption with ipsec using PSK, encryption with ipsec using cert. In my understanding, if we support all changes, we need to revert 3 possible configurations installed by previous modes. Perhaps some changes are supported but I don't see such cleanup is organized well and perhaps cleanup is not really clean even if some of them work.

Anyway even if we want to support such changes, the cleanup of IPSec with Certs should not be executed under the condition about its own configuration, for example:
Instead of:
```
if wireguard {
  initWireguard()
  cleanupIPsecCert()
  cleanupIPsecPSK()
} else if ipsecCert {
  initIPsecCert()
  cleanupWireguard()
  cleanupIPsecPSK()
} else if ipsecPSK {
  initIPsecPSK()
  cleanupWireguard()
  cleanupIPsecCert()
} else {
  cleanupWireguard()
  cleanupIPsecCert()
  cleanupIPsecPSK()
}
```
It should do:
```
if wireguard {
  initWireguard()
} else {
  cleanupWireguard()
}
if ipsecCert {
  initIPsecCert()
} else {
  cleanupIPsecCert()
}
if ipsecPSK {
  initIPsecPSK()
} else {
  cleanupIPsecPSK()
}
```

Reply:
I need to figure out a clean way to do this as suggested.

Reply:
I changed the cleanup logic as suggested. The cleanup is necessary for this version as we need to test both "psk" and "cert" modes for IPsec encryption.